extension/demo/src/bg/background.js

```diff
@@ -23,44 +24,59 @@ function onPopupOpenApps () {
 
 async function onToken (data) {
   const { accessToken, subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY] = {
-    auth: {
-      accessToken: accessToken,
-      subscription: subscription
-    }
-  }
-  await chrome.storage.local.set(payload)
-  const isAuthenticated = !chrome.runtime.lastError
+  const isAuthenticated = await storage.set({ accessToken, subscription })
   sendMessage('popup-set-view', {
     isAuthenticated
   })
+
   return {
     isAuthenticated
   }
 }
 
+function timeToLive (expiresAt) {
+  return expiresAt - (Math.floor(Date.now() / 1000) + 3600)
+}
+
+async function updatePayload () {
+  const payload = await storage.get(PAYLOAD)
+  if (!payload) {
+    const response = await api.fetchPayload()
+    if (response.hasOwnProperty(PAYLOAD)) { // fetch return might need to go through .json()?
+      let { expiresAt, payload } = response
+      await storage.set({ expiresAt, payload })
+    }
+  } else {
+    const ttl = await storage.get(TTL)
+    if (ttl) {
+      setTimeout(updatePayload, timeToLive(ttl))
+    }
+  }
+}
+
 async function onSubscription (data) {
   const { subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY]['auth']['subscription'] = subscription
-  await chrome.storage.local.set(payload)
-  chrome.storage.local.get([STORAGE_KEY], results => {
-    console.log(results)
-    return results
-  })
+  await storage.set({ subscription })
 }
 
 async function onPopupCheckAuth () {
-  await chrome.storage.local.get([STORAGE_KEY], results => {
-    sendMessage('popup-set-view', {
-      isAuthenticated: !!results[STORAGE_KEY]
-    })
+  const accessToken = await storage.get(ACCESS_TOKEN)
+  sendMessage('popup-set-view', {
+    isAuthenticated: !!accessToken
   })
 }
 
-addListener('token', onToken)
-addListener('subscription', onSubscription)
-addListener('popup-check-auth', onPopupCheckAuth)
-addListener('popup-trigger-auth', onPopupTriggerAuth)
-addListener('popup-open-apps', onPopupOpenApps)
+async function init () {
+  addListener('token', onToken)
+  addListener('subscription', onSubscription)
+  addListener('popup-check-auth', onPopupCheckAuth)
+  addListener('popup-trigger-auth', onPopupTriggerAuth)
+  addListener('popup-open-apps', onPopupOpenApps)
+
+  const accessToken = await storage.get(ACCESS_TOKEN)
+  if (accessToken) {
+    await updatePayload()
+  }
+}
+
+await init()
```

Comment:
This won't work. You can't use `await` outside an async function. Also there's not really any reason to wait here, there's nothing we need to block.

I would probably just make this a self invoking function instead:

```
;(async => {
  // ...
})()
```